app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeight.kt

```diff
@@ -16,17 +18,24 @@ class AddMaxHeight : OsmElementQuestType<MaxHeightAnswer> {
           or amenity = parking_entrance and parking ~ underground|multi-storey
         )
         and !maxheight and !maxheight:physical
-    """.toElementFilterExpression()}
+    """.toElementFilterExpression() }
 
     private val wayFilter by lazy { """
         ways with
         (
           highway ~ motorway|motorway_link|trunk|trunk_link|primary|primary_link|secondary|secondary_link|tertiary|tertiary_link|unclassified|residential|living_street|track|road
           or (highway = service and access !~ private|no and vehicle !~ private|no)
         )
-        and (covered = yes or tunnel ~ yes|building_passage|avalanche_protector)
         and !maxheight and !maxheight:physical
-    """.toElementFilterExpression()}
+    """.toElementFilterExpression() }
+
+    private val tunnelFilter by lazy { """
+        ways with highway and (covered = yes or tunnel ~ yes|building_passage|avalanche_protector)
+    """.toElementFilterExpression() }
+
+    private val bridgeFilter by lazy { """
+        ways with highway and bridge and layer
```

Comment:
Maybe catching also pipelines over roads would be nice? https://overpass-turbo.eu/s/ZGn

https://www.openstreetmap.org/way/188354829#map=17/50.00945/19.88687 https://www.openstreetmap.org/way/83175684

Reply:
Also, skipping `highway=proposed` is a good idea (maybe just match to all roads + `steps` `footway` `path` `cycleway`) to avoid discovering more things that should be skipped.

Skipping `highway=construction` is likely a good idea, often such line is correctly mapped at but bridge is not yet constructed.

Reply:
Wouldn't catch your pipeline over there because `layer=1` is missing though

Reply:
Also, the location of the pipeline should be "overhead", not "overground", according to the wiki

Reply:
Ok, worked in your suggestions